File: 351003/Guzaev/app/controllers/comment_controller.py
from fastapi import APIRouter, status, Response
from typing import List
from app.dtos.comment_dto import CommentRequestTo, CommentResponseTo
from app.services.comment_service import CommentService

router = APIRouter(prefix="/api/v1.0/comments", tags=["comments"])
service = CommentService()

@router.post("", status_code=status.HTTP_201_CREATED, response_model=CommentResponseTo)
def create_comment(dto: CommentRequestTo):
    return service.create(dto)

@router.get("", response_model=List[CommentResponseTo])
def get_comments():
    return service.get_all()

@router.get("/{id}", response_model=CommentResponseTo)
def get_comment(id: int):
    return service.get_by_id(id)

@router.put("/{id}", response_model=CommentResponseTo)
def update_comment(id: int, dto: CommentRequestTo):
    return service.update(id, dto)

@router.delete("/{id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_comment(id: int):
    service.delete(id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)

File: 351004/Kadushko/Lab1/restApi/src/main/java/com/example/restApi/controllers/CommentController.java
package com.example.restApi.controllers;

import com.example.restApi.dto.request.CommentRequestTo;
import com.example.restApi.dto.response.CommentResponseTo;
import com.example.restApi.services.CommentService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/comments")
public class CommentController {

    private final CommentService commentService;

    public CommentController(CommentService commentService) {
        this.commentService = commentService;
    }

    @GetMapping
    public ResponseEntity<List<CommentResponseTo>> getAll() {
        return ResponseEntity.ok(commentService.getAll());
    }

    @GetMapping("/{id}")
    public ResponseEntity<CommentResponseTo> getById(@PathVariable Long id) {
        return ResponseEntity.ok(commentService.getById(id));
    }

    @PostMapping
    public ResponseEntity<CommentResponseTo> create(@Valid @RequestBody CommentRequestTo request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(commentService.create(request));
    }

    @PutMapping
    public ResponseEntity<CommentResponseTo> update(@Valid @RequestBody CommentRequestTo request) {
        return ResponseEntity.ok(commentService.update(request));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Long id) {
        commentService.delete(id);
        return ResponseEntity.noContent().build();
    }
}
